### PR TITLE
ChipTool: Make the ShowUsage output grouped by clusters

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

ChipTool help is hard to read, and it will be harder and harder when adding more commands.

 #### Summary of Changes
* Prettify the output by making per cluster commands group

### CODE REVIEW

File: examples/chip-tool/commands/common/Utils.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ */
+
+#ifndef __CHIPTOOL_UTILS_H__
+#define __CHIPTOOL_UTILS_H__
+
+#include <memory>
+#include <vector>
+
+class Command;
```
Comment (andy31415) [inline examples/chip-tool/commands/common/Utils.h:25]: should we #include instead?

Reasoning being: these structures/methods would not be usable without a definision of Command anyway and #include would prevent us from creating some circular dependencies.

as an aside, I am somewhat concerned about anything named 'helpers' or 'utils' (not obvious what they are when #included, end up being a mixed bad of everything inside). Maybe we could make these something else (CommandBuilders, CommandLists or just Lists).

File: examples/chip-tool/commands/echo/Commands.h [hunk 2]
```diff
@@ -35,8 +35,14 @@ class EchoBle : public EchoCommand
 
 void registerCommandsEcho(Commands & commands)
 {
-    commands.Register(make_unique<Echo>());
-    commands.Register(make_unique<EchoBle>());
+    const char * clusterName = "";
```
Comment (andy31415) [inline examples/chip-tool/commands/echo/Commands.h:38]: should this be echo?

**Review by andy31415 — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

File: examples/chip-tool/commands/echo/Commands.h (hunk 2 above)
Comment (vivien-apple) [reply examples/chip-tool/commands/echo/Commands.h:38]: Not really. This is an empty name until the echo command is really part of a Cluster. See https://github.com/project-chip/connectedhomeip/pull/3020

File: examples/chip-tool/commands/common/Utils.h (hunk 1 above)
Comment (vivien-apple) [reply examples/chip-tool/commands/common/Utils.h:25]: I got your point about the name. I will rename it to something more specific.

Comment (vivien-apple) [reply examples/chip-tool/commands/common/Utils.h]: I removed the file and move the helpers directly into Commands.h.

**Review by rwalker-apple — APPROVED** [review]

File: examples/chip-tool/commands/common/Commands.cpp [hunk 3]
```diff
@@ -21,9 +21,12 @@
 #include "Command.h"
 #include <string>
 
-void Commands::Register(std::unique_ptr<Command> command)
+void Commands::Registers(const char * clusterName, commands_list commandsList)
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/common/Commands.cpp:24]: Should still be called "Register", right?

File: examples/chip-tool/commands/common/Commands.cpp [hunk 4]
```diff
@@ -52,17 +55,19 @@ CHIP_ERROR Commands::RunCommand(ChipDeviceController & dc, NodeId remoteId, int
 
     VerifyOrExit(argc > 1, err = CHIP_ERROR_INVALID_ARGUMENT);
 
-    for (size_t i = 0; i < commands.size(); i++)
+    for (auto & cluster : clusters)
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/common/Commands.cpp:58]: This is going to enumerate "clusters" in a random order in general, right?  Is that ok for our purposes?  I'd think we want something stable here...

Comment (vivien-apple) [reply examples/chip-tool/commands/common/Commands.cpp:58]: I thought it was going to be in the order of the key (which is a string).

Comment (bzbarsky-apple) [reply examples/chip-tool/commands/common/Commands.cpp:58]: Ah, ok.  In that case, great.

**Pull request merged**